### ISSUE TITLE
[BUG] Juno doesn't re-use pre-compilation from different IDEs

### ISSUE THREAD

**Issue by Datseris** [JunoLab/Juno.jl#589]
Whenever I use some package in Juno, that I have in dev form, Juno will precompile it. This sounds reasonable, but the moment I switch development environment, and use e.g. REPL or Jupyter, everything that Juno precompiled gets again precompiled. The same thing happens vice versa of course.

This doesn't make sense, why does it happen? 

Versions: Windows 10, Julia 1.4, Julia packages latest stable, Atom packages latest stable.

**Comment by pfitzseb** [comment]
Probably because you're actually getting a different "environment" in Juno vs the REPL. And this happens because Juno already loads a bunch of packages, so things break if your Manifest specifies a different version.

**Comment by Datseris** [comment]
Can you please help me identify the different "environments" so that I can configure Juno to use the same? Because I only have Julia 1.4 and therefore assume that this is my only environment...

**Comment by pfitzseb** [comment]
This is going to happen whenever you do a `pkg> activate Foo` where the environment Foo specifies a version of one of Atom.jl's dependencies different from the one in Atom's Manifest.toml. 

If you're only using the global environment then it's a different issue.

**Comment by Datseris** [comment]
This happens for both Global as well as Specific environments. So it does happen on `activate Foo`, but it _also_ happens if I use just the global environment and just do `using ThisPackage` where `ThisPackage` is out for dev.

**Comment by Datseris** [comment]
> different from the one in Atom's Manifest.toml.

I am confused by this statement. Packages don't have a Manifest file, or they *shouldn't* have, because the PRoject.toml is what specifies the dependencies, not the Manifest.toml. Where is this file so I can have a look?

**Comment by pfitzseb** [comment]
Yeah, that should've been "...different from the one in the global environment". Are you sure you're using the same Julia version everywhere?

**Comment by Datseris** [comment]
Yeah, I am certain I use the same Julia version everywhere.
